independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java

```diff
@@ -74,6 +74,12 @@
 
     @Override
     public void fire(T event) {
+        Type resolvedType = Types.getCanonicalType(getEventType(event.getClass()));
```

Comment:
This check is IMO superfluous because of the logic from the `EventImpl.getEventType()` which is called when a notifier is created.

Reply:
Or at least we should improve the check in `EventImpl.getEventType()` instead...

Reply:
You're right, using `getEventType()` to get to the event `Type` should be enough. Will verify.

Reply:
My point is that we call the `getEventType()` method  when a new notifier is created and so we can just add the `if (Types.containsTypeVariable(eventType))` check at the end of the `getEventType()` method, or?